pandas/io/common.py

```diff
@@ -553,8 +553,10 @@ def get_handle(
     Returns the dataclass IOHandles
     """
     # Windows does not default to utf-8. Set to utf-8 for a consistent behavior
+    encoding_not_specified = False
```

Comment:
instead why don't you
```
encoding_passed, encoding = encoding, encoding or 'utf-i'
```

and then you can test `encoding_passed is not None` its the same as you have but i think a bit more natural